sdk/identity/azure-identity/azure/identity/_credentials/environment.py

```diff
@@ -76,6 +81,21 @@ def __init__(self, **kwargs):
                 **kwargs
             )
 
+        if self._credential:
+            _LOGGER.info("Environment is configured for %s", self._credential.__class__.__name__)
```

Comment:
Just a question since I'm not familiar enough with how the credentials are set in python, do you log environment configurations for other credentials that are affected when there are environment variables present? For instance, in managed identity, if client ID is set, that affects the credential.

Reply:
No, in this PR only `EnvironmentCredential` logs environment variables. Do we intend to log that a particular variable is set everywhere it's used?

Reply:
I think we did settle on the fact that this is an important logging scenario, since env vars do impact the request for credentials when they are present. I would say that if possible we should only log them once per total log output. In Go, I have not quite reached that, I log them once per credential that is instantiated.